src/App.tsx

```diff
@@ -1,24 +1,182 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useState,
+  ChangeEvent,
+  FormEvent,
+} from 'react';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { ErrorPanel } from './components/ErrorPanel';
+
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filters';
+
+import { getTodos, addTodo, removeTodo } from './api/todos';
+
+const USER_ID = 6160;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [isEditing] = useState(false);
+  // const [hasChangesInTodos, setHasChangesInTodos] = useState(0);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [selectedFilter, setSelectedFilter] = useState<Filters>(Filters.All);
+  const [todoTitleToAdd, setTodoTitleToAdd] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [idsOfLoadingTodos, setIdsOfLoadingTodos] = useState<number[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const onNewTitleAdd = (event: ChangeEvent<HTMLInputElement>) => {
+    setTodoTitleToAdd(event.target.value);
+  };
+
+  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (todoTitleToAdd) {
+      setIsInputDisabled(true);
+      const todoToAdd: Todo = {
+        id: 0,
+        userId: USER_ID,
+        title: todoTitleToAdd,
+        completed: false,
+      };
+
+      setTempTodo(todoToAdd);
+
+      addTodo(todoToAdd)
+        .then((newTodo) => {
+          setTodos([...todos, newTodo]);
+          setFilteredTodos([...todos, newTodo]);
+          setTempTodo(null);
+          setIsInputDisabled(false);
+        })
+        .catch(() => {
+          setError('Unable to add a todo');
+        });
+      setTodoTitleToAdd('');
+    } else {
+      setError('Title can\'t be empty');
+    }
+  };
```

Comment:
looks really messy. You can validate the title at the very beginning for make it look a bit clearer
```suggestion
  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    
    if (!todoTitle.length) {
      setError('Title can\'t be empty');
    }
    
      setIsInputDisabled(true);
      
      const todoToAdd: Todo = {
        id: 0,
        userId: USER_ID,
        title: todoTitleToAdd,
        completed: false,
      };

      setTempTodo(todoToAdd);

      addTodo(todoToAdd)
        .then((newTodo) => {
          setTodos([...todos, newTodo]);
          setFilteredTodos([...todos, newTodo]);
          setTempTodo(null);
          setIsInputDisabled(false);
        })
        .catch(() => {
          setError('Unable to add a todo');
        });
        
      setTodoTitleToAdd('');
  };
```